orchestrator/types.py

```diff
@@ -148,6 +148,16 @@ def is_list_type(t: Any, test_type: Optional[type] = None) -> bool:
     False
     >>> is_list_type(Literal[1,2,3])
     False
+    >>> is_list_type(List[Union[str, int]])
+    True
+    >>> is_list_type(List[Union[str, int]], Union[str, int])
+    True
```

Comment:
Type checking with a union that has args in different order is still not behaving as expected. (see old comment https://github.com/workfloworchestrator/orchestrator-core/pull/90#discussion_r801682036)

Any idea if it can be fixed? Otherwise we can put a note/todo in the docstring about this limitation.

Reply:
It's a shortcoming of `is_of_type()` which first does an identity check, which of course passes for this case

```

In [16]: is_of_type(Union[str, int], Union[str, int])
Out[16]: True
```

But when changing the order the identity check returns false, so it does a `issubclass` check which raises a (catched) exception
```
In [17]: is_of_type(Union[str, int], Union[int,str])
Out[17]: False
```

The error being

```
In [19]: issubclass(Union[str, int], Union[int, str])
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [19], in <module>
----> 1 issubclass(Union[str, int], Union[int, str])

File ~/.pyenv/versions/3.9.10/lib/python3.9/typing.py:722, in _BaseGenericAlias.__subclasscheck__(self, cls)
    721 def __subclasscheck__(self, cls):
--> 722     raise TypeError("Subscripted generics cannot be used with"
    723                     " class and instance checks")

TypeError: Subscripted generics cannot be used with class and instance checks
```

A quick google shows a few more people have this question but there isn't an easy solution. Looks like this library could solve the problem but not sure if we want to add it just for this https://github.com/ilevkivskyi/typing_inspect

Reply:
browsed a bit through the typing_inspect issues and PR's: looks solid and somewhat maintained.

I can confirm that with the last changes the unit test of the SURF orchestrator pass.